File: server/src/main/java/com/krillsson/sysapi/core/monitoring/monitors/MemoryMonitor.java
package com.krillsson.sysapi.core.monitoring.monitors;

import com.krillsson.sysapi.core.domain.system.SystemLoad;
import com.krillsson.sysapi.core.monitoring.Monitor;
import com.krillsson.sysapi.core.monitoring.MonitorType;

import java.time.Duration;

public class MemoryMonitor extends Monitor {
    public MemoryMonitor(String id, Duration inertia, double threshold) {
        super(id, inertia, threshold);
    }

    @Override
    protected double value(SystemLoad systemLoad) {
        return systemLoad.getMemory().getAvailable();
    }

    @Override
    protected boolean isOutsideThreshold(double value) {
        return value < threshold();
    }

    @Override
    protected MonitorType type() {
        return MonitorType.MEMORY_SPACE;
    }
}

File: server/src/main/java/com/krillsson/sysapi/core/monitoring/monitors/MemoryMonitor.kt
package com.krillsson.sysapi.core.monitoring.monitors

import com.krillsson.sysapi.core.domain.monitor.MonitorConfig
import com.krillsson.sysapi.core.domain.monitor.MonitoredValue
import com.krillsson.sysapi.core.domain.monitor.toNumericalValue
import com.krillsson.sysapi.core.monitoring.Monitor
import com.krillsson.sysapi.core.monitoring.MonitorMetricQueryEvent
import java.util.*

class MemoryMonitor(override val id: UUID, override val config: MonitorConfig<MonitoredValue.NumericalValue>) : Monitor<MonitoredValue.NumericalValue>() {
    override val type: Type = Type.MEMORY_SPACE

    override fun selectValue(event: MonitorMetricQueryEvent): MonitoredValue.NumericalValue {
        return event.load().memory.availableBytes.toNumericalValue()
    }

    override fun isPastThreshold(value: MonitoredValue.NumericalValue): Boolean {
        return value < config.threshold
    }
}